sktree/tree/_unsup_criterion.pyx

```diff
@@ -398,3 +399,154 @@ cdef class TwoMeans(UnsupervisedCriterion):
         self.n_node_samples = end - start
         self.start = start
         self.end = end
+
+cdef class FastBIC(UnsupervisedCriterion):
```

Comment:
Subclass two means criterion instead to leverage reuse of the sum of squares function.